batch_job/src/work_queue_factory.c

```diff
@@ -316,14 +317,33 @@ static void set_worker_resources_options( struct batch_queue *queue )
 	resource_args = xxstrdup(buffer_tostring(&b));
 	buffer_free(&b);
 }
-
+static char* fetch_end_of_path(char* path){
+	size_t len= strlen(path);
+	size_t slash_pos = 0;
+	size_t i=0;
+	for(i=len-1; i > -1; i-- ){
+		if(path[i] == '/'){
+			slash_pos = i;
+			break;
+		}
+	}
+	char* new_str = malloc(sizeof(char)*(len-slash_pos)+1);
+	for(i=0; i< (len-slash_pos); i++){
+		new_str[i] = path[slash_pos + i];
+	}
+	new_str[len-slash_pos] = '\0';
+	return new_str;
+}
 static int submit_worker( struct batch_queue *queue )
 {
 	char *cmd;
+	char *tmp = fetch_end_of_path(worker_prog);
+	char *worker = (worker_prog != NULL) ? tmp : string_format("./work_queue_worker");
```

Comment:
There's some inconsistency here?
```
const char *worker = worker_command ? path_basename(worker_command) : "work_queue_worker";
```
In one part of the command below `./%s` is userd, in the other `%s`. Why not change them both to `./%s` and drop the `./` here in front of `work_queue_worker`?